sycl/doc/design/spirv-extensions/SPV_INTEL_joint_matrix.asciidoc

```diff
@@ -296,6 +297,36 @@ specifying the <<Memory_Operands,*memory operand*>> *None*. +
 'Memory Access'
 |=====
 
+==== 3.37.12. Composite Instructions
+
+Modify *OpVectorExtractDynamic* and *OpVectorInsertDynamic* to accept
+<<OpTypeJointMatrixINTEL,*OpTypeJointMatrixINTEL*>> as the 'Vector' operand.
+In this case the instructions operate on an implicit vector which repesents
+part of the joint matrix and holds components owned by the current work-item.
+If the 'index' operand of these instructions exceeds the value returned by
+<<OpJointMatrixWorkItemLengthINTEL,*OpJointMatrixWorkItemLengthINTEL*>>,
+behavior is undefined.
+
+[cols="1,1,3*3",width="100%"]
+|=====
+4+|[[OpJointMatrixWorkItemLengthINTEL]]*OpJointMatrixWorkItemLengthINTEL* +
+ +
+Return number of components owned by the current work-item in a joint matrix. +
+ +
+'Result Type' must be an unsigned integer type. +
```

Comment:
At some point we may want to clarify that the result type must be an unsigned integer-type **scalar**, unless we think there is a use-case for a vector result type.

Do we want to support arbitrary integer sizes?  I was comparing this instruction to **OpSizeOf** just to understand the differences, and for what it's worth **OpSizeOf** says:

> Result Type must be a 32-bit integer type scalar.

Reply:
Thanks Ben! I think, that Result should fir in one SPIR-V Word aka be 32-bit integer. I'll update the spec.